### PR TITLE
jj: update to 0.9.0

### PR DESCRIPTION
**Pull request by ognevnydemon**
notable changes:
- build with all features
- add split package for docs
- add completions for shells

### CODE REVIEW

File: mingw-w64-jj/PKGBUILD [hunk 1]
```diff
@@ -15,53 +15,64 @@ msys2_references=(
 url="https://github.com/martinvonz/jj"
 license=('spdx:Apache-2.0')
 makedepends=("${MINGW_PACKAGE_PREFIX}-rust" "git")
-source=("git+https://github.com/martinvonz/jj#commit=${_commit}")
+source=("${_realname}::git+${url}#tag=v${pkgver}")
```
Comment (mmuetzel) [inline mingw-w64-jj/PKGBUILD:18]: Would it make sense to build from the release tarball instead?
``` suggestion
source=("${_realname}-${pkgver}.tar.gz::https://github.com/martinvonz/jj/archive/refs/tags/v${pkgver}.tar.gz")
```
(Untested)

Comment (ognevnydemon) [reply mingw-w64-jj/PKGBUILD:18]: in this way we also install source code. the only difference is that it's already signed and .git directory is included

Comment (mmuetzel) [reply mingw-w64-jj/PKGBUILD:18]: > in this way we also install source code.

Source tarballs usually contain source code. Is that different for that project?

> .git directory is included
Do you need that directory?

Comment (ognevnydemon) [reply mingw-w64-jj/PKGBUILD:18]: > Source tarballs usually contain source code. Is that different for that project?
no
> Do you need that directory?
no

I just left source as it was. IMO there is no reason to change the way how source is downloaded

Comment (mmuetzel) [reply mingw-w64-jj/PKGBUILD:18]: If possible try to stay close to the PKGBUILD templates to keep maintenance easy.

In this case, the closest template is probably:
https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-PKGBUILD-templates/PKGBUILD.cargo-tarball.sh

Comment (ognevnydemon) [reply mingw-w64-jj/PKGBUILD:18]: anyway this is note essential. if I can leave this as is, I'll leave it as is

**Comment by lazka** [comment]
either build a tarball or a commit, git tags can be changed without us noticing.

**Comment by ognevnydemon** [comment]
> either build a tarball or a commit, git tags can be changed without us noticing.

oh, ok. thanks for the answer. will change today

**Comment by ognevnydemon** [comment]
> oh, ok. thanks for the answer. will change today

done

**Comment by ognevnydemon** [comment]
oops, binary was removed...

**Comment by ognevnydemon** [comment]
any other objections?

**Comment by lazka** [comment]
I find the split package for 180kb docs questionable, but ymmv :)

**Comment by ognevnydemon** [comment]
> I find the split package for 180kb docs questionable, but ymmv :)

these docs are actually compiled with mkdocs, but IMO it will cool to let someone to read them using good reader

**Pull request merged**